SubsoniciousKit/SubsoniciousKit/Authentication/AuthenticationManager.swift

```diff
@@ -12,19 +12,53 @@ import Foundation
 public class AuthenticationManager: ObservableObject {
 
     private let authenticationService: AuthenticationService
+    private let serverPersistenceManager: ServerPersistenceManager
     @Published public private(set) var result: Result<SubsonicResponse, Error>?
+    private var server: Server?
     private var cancellables: Set<AnyCancellable> = []
 
-    public init(authenticationService: AuthenticationService) {
+    public init(authenticationService: AuthenticationService,
+                serverPerstistenceManager: ServerPersistenceManager) {
         self.authenticationService = authenticationService
+        self.serverPersistenceManager = serverPerstistenceManager
+
+        bindResult()
+    }
+
+    private func bindResult() {
+        $result
```

Comment:
is this called only from authenticate ? why split the stream with this binding in addition to authenticate method ?

Reply:
Because `authenticate(with:)` cares about the authentication itself while the binding aims to persist the server when the result is a success.
What do you think?

Reply:
What I understand of the current code =
Authenticate(with:) is called => Web service call => put the result in a var => with the listening of this variable, persist it. 
every time, and there is no case of using the result for something else.
I think the separation is not needed and tend to make the code more complex.
Why not Authenticate(with:) is called => Web service call => persist it. 
in one thread.
You could cut if needed with share, but that's not the case here, you just want to continue the work thread.

Reply:
actually I use the result to update the UI (redirect the user to next screen or display an error message) :)